src/http/fetch.js

```diff
@@ -0,0 +1,30 @@
+export default function(fetch) {
+    return (config) => {
+        const url = config.url;
+        delete config.url;
+
+        if (config.data) {
+            config.body = /application\/json/.test(config.headers['Content-Type']) ? JSON.stringify(config.data) : config.data;
+            delete config.data;
+        }
+
+        return fetch(url, config)
+            .then((response) => {
+                if (response.status >= 200 && response.status < 300) {
+                    return response;
+                }
+
+                const error = new Error(response.statusText);
+                error.response = response;
+                throw error;
+            })
+            .then((response) => {
+                return response.json().then((json) => {
+                    return {
+                        data: json,
+                        statusCode: response.status,
```

Comment:
Shouldn't we also add the response headers? It may be useful for instance in case of a paginated `GET /entities` query. The total number of elements may be fetched thanks to a `X-Total-Count` header.